test/e2e/suite/trustpolicy/trustpolicy_test.go

```diff
@@ -0,0 +1,467 @@
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			artifact := GenerateArtifact("e2e-with-expired-cert", "")
+
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("audit_trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2EConfigPath, "localkeys", "expired_e2e.crt")))
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticTimestamp was set to \"log\"",
+					"error: certificate \"O=Internet Widgits Pty Ltd,ST=Some-State,C=AU\" is not valid anymore, it was expired",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test audit signatureVerification level with invalid authenticity", func() {
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("audit_trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2ELocalKeysDir, "new_e2e.crt")))
+
+			// the artifact signed with a different cert from the cert in
+			// trust store.
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticity was set to \"log\"",
+					"signature is not produced by a trusted signer",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test audit signatureVerification level with invalid integrity", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("audit_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-invalid-signature", "")
+
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchErrKeyWords("integrity validation failed",
+					VerifyFailed)
+		})
+	})
+
+	It("test skip signatureVerification level with invalid integrity", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("skip_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-invalid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Trust policy is configured to skip signature verification")
+		})
+	})
+
+	It("test override strict signatureVerification level with Expiry set to log", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("override_strict_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-expired-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("digital signature has expired",
+					"expiry was set to \"log\"",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test override strict signatureVerification level with Authentic timestamp set to log", func() {
```

Comment:
Do we have an example of "test override strict signatureVerification level with Integrity set to log"? Since we do not allow override of Integrity, we should expect an error for this example.

Reply:
Added.